ddtrace/internal/writer.py

```diff
@@ -392,3 +449,6 @@ def run_periodic(self):
         self.flush_queue(raise_exc=False)
 
     on_shutdown = run_periodic
+
+
+TraceWriter.register(AgentWriter)
```

Comment:
Why can't AgentWriter inherit from `TraceWriter` ?

Reply:
The ABC doesn't evaluate the MRO with multiple inheritance

```python
>>> from ddtrace.internal.writer import AgentWriter
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kyle.verhoog/dev/dd-trace-py/ddtrace/__init__.py", line 20, in <module>
    tracer = Tracer()
  File "/Users/kyle.verhoog/dev/dd-trace-py/ddtrace/tracer.py", line 126, in __init__
    self.configure(
  File "/Users/kyle.verhoog/dev/dd-trace-py/ddtrace/tracer.py", line 289, in configure
    self.writer = AgentWriter(
TypeError: Can't instantiate abstract class AgentWriter with abstract methods is_alive, join, shutdown, stop
```

Reply:
Oh, I had the class ordering the wrong way round which makes sense. Updating!